components/src/core/components/InputField/InputField.vue

```diff
@@ -101,26 +79,59 @@ export default defineComponent({
     type: {
       type: String,
       default: TYPE_INPUT,
-      validator: function(value: Types) {
+      validator: (value: Types) => {
         return TYPES.indexOf(value) !== -1;
       },
     },
     errors: {
       type: String,
       default: TYPE_INPUT,
-      validator: function(value: Types) {
+      validator: (value: Types) => {
         return TYPES.indexOf(value) !== -1;
       },
     },
+    rules: {
+      type: Array as PropType<any>,
+      default: () => [],
+    },
+  },
+
+  setup(props) {
+    const modelValue = toRef(props, 'modelValue');
+    const {
+      form,
+      validate,
+      hasError,
+      message,
+      getProcessingState,
+      setProcessingState,
+    } = useField({
+      fieldLabel: props.label,
+      rules: props.rules,
+      modelValue,
+    });
+
+    const onBlur = async () => {
```

Comment:
Previously the `Inputfield` was executing validations when `modelvalue` changed. but here i have used `blur` event since in asynchronous validations, it will create many requests to back-end on input. need a `debounce` behaviour in validation rules.

Reply:
Changed the validation executions back to `onChange` `modelvalue` as many other synchronous validators depend on it. due to this any async validator we may introduce to the system must be debounced before introducing. Added a [promise based debounce function](https://github.com/orangehrm/oxd/pull/47/files#diff-94e8b9fcb59f2aa6d054e044aafd7bcf87d47fafd57219d074e024ac6938a9fb) for our uses.